apps/api/src/services/perps-data-processor.service.ts

```diff
@@ -0,0 +1,730 @@
+
+  /**
+   * Process all agents in a competition
+   * @param competitionId Competition ID
+   * @param provider The perps data provider to use
+   * @returns Batch processing results with account summaries
+   */
+  async processCompetitionAgents(
+    competitionId: string,
+    provider: IPerpsDataProvider,
+  ): Promise<
+    BatchPerpsSyncResult & {
+      accountSummaries: Map<string, PerpsAccountSummary>;
+      agents: Array<{ agentId: string; walletAddress: string }>;
+    }
+  > {
+    if (!provider) {
+      throw new Error("[PerpsDataProcessor] Provider is required");
+    }
+
+    serviceLogger.info(
+      `[PerpsDataProcessor] Processing all agents for competition ${competitionId}`,
+    );
+
+    // Get competition agents
+    const agentIds = await getCompetitionAgents(competitionId);
+
+    if (agentIds.length === 0) {
+      serviceLogger.info(
+        `[PerpsDataProcessor] No agents found for competition ${competitionId}`,
+      );
+
+      return {
+        successful: [],
+        failed: [],
+        accountSummaries: new Map(),
+        agents: [],
+      };
+    }
+
+    // Fetch agent details
+    const agents = await findAgentsByIds(agentIds);
+
+    // Filter out agents without wallet addresses and map to required format
+    const agentData = agents
+      .filter(
+        (agent): agent is typeof agent & { walletAddress: string } =>
+          agent.walletAddress !== null,
+      )
+      .map((agent) => ({
+        agentId: agent.id,
+        walletAddress: agent.walletAddress,
+      }));
+
+    if (agentData.length < agents.length) {
+      const missingCount = agents.length - agentData.length;
+      serviceLogger.warn(
+        `[PerpsDataProcessor] ${missingCount} agents have no wallet address and will be skipped`,
+      );
+    }
+
+    return this.processBatchAgentData(agentData, competitionId, provider);
+  }
+
+  /**
+   * Get competition configuration for perps
+   * @param competitionId Competition ID
+   * @returns Competition configuration or null if not found
+   */
+  async getCompetitionConfig(competitionId: string) {
+    return getPerpsCompetitionConfig(competitionId);
+  }
+
+  /**
+   * Validate competition is a perps competition
+   * @param competitionId Competition ID
+   * @returns True if competition is configured for perps
+   */
+  async isPerpsCompetition(competitionId: string): Promise<boolean> {
+    const competition = await findCompetitionById(competitionId);
+
+    if (!competition) {
+      serviceLogger.warn(
+        `[PerpsDataProcessor] Competition ${competitionId} not found`,
+      );
+      return false;
+    }
+
+    return competition.type === "perpetual_futures";
+  }
+
+  /**
+   * High-level orchestration method for perps competitions
+   * Handles data sync and optional self-funding monitoring
+   * @param competitionId Competition ID
+   * @returns Combined results from sync and monitoring
+   */
+  async processPerpsCompetition(competitionId: string): Promise<{
```

Comment:
This is the most important method - the one that's integrated into the chron script in https://github.com/recallnet/js-recall/pull/1219 , responsible for 
- Getting positions and summaries from the perps data provider (in this case, Symphony)
- Running self-funding monitoring alerts
- Persisting the data